dotcom-rendering/scripts/islands/descriptions.mjs

```diff
@@ -0,0 +1,172 @@
+// @ts-check
+
+import { readdir, readFile, writeFile } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import rehypeAutolinkHeadings from 'rehype-autolink-headings';
+import rehypeSlug from 'rehype-slug';
+import rehypeStringify from 'rehype-stringify';
+import remarkParse from 'remark-parse';
+import remarkRehype from 'remark-rehype';
+import { unified } from 'unified';
+import * as zod from 'zod';
+
+/** @type {(name: string) => RegExp} */
+export const getRegExForIsland = (name) =>
+	new RegExp(
+		'\n/\\*\\*\n?' +
+			'((?: \\*.*\n)+)' +
+			' \\*\\/' +
+			`\nexport const ${name} =`,
+		'm',
+	);
+
+const componentsDirectory = resolve(
+	fileURLToPath(import.meta.url),
+	'../../../src/web/components',
+);
+
+const PREFIX = 'window.chartData = ';
+
+const chunk = zod.object({
+	label: zod.string(),
+	isAsset: zod.boolean(),
+	statSize: zod.number(),
+	parsedSize: zod.number(),
+	gzipSize: zod.number(),
+});
+
+/** Sorted by gzipSize */
+const bundleReport = await readFile(
+	resolve(
+		fileURLToPath(import.meta.url),
+		'../../../dist/browser.modern-bundles.html',
+	),
+	'utf-8',
+)
+	.then((file) =>
+		file
+			.split('\n')
+			.find((line) => line.trim().startsWith(PREFIX))
+			?.trim(),
+	)
+	.then((bundle_data) =>
```

Comment:
There's a few `snake_case` variables in here,  is this because our linter isn't setup for `mjs`?

Reply:
I don’t think we lint on variable names at all… it’s a convention not a rule 🤷

Reply:
Just FYI, AR introduced some linting for naming conventions a while back. That won't affect this, but just in case you're interested.

https://github.com/guardian/dotcom-rendering/blob/c56037a6690220c0228db852762a8d0c5281c26a/apps-rendering/config/.eslintrc.js#L49-L85